test/unit/controllers/BusinessDetailsRecoveryControllerSpec.scala

```diff
@@ -178,6 +169,43 @@ class BusinessDetailsRecoveryControllerSpec extends ControllerSpec with BeforeAn
       }
     }
 
+  private def assertAndTestThrowsExceptionForInvalidLocationOrganisation: Unit =
+    "throw IllegalStateException when passing invalid location" in {
```

Comment:
Whilst I'm happy to see you testing this case that wasn't before, please create a technical debt ticket to assess this exception and if it's needed/we can improve the program flow.